### PR TITLE
Add FIPS Indicator for ECDH cofactor.

### PR DESCRIPTION
**Pull request by slontis**
FIPS KAS requires use of ECC CDH.

The EC 'B' and 'K' curves have a cofactor that is not 1, and this MUST be multiplied by the private key when deriving the shared secret.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: providers/common/include/prov/proverr.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * Generated by util/mkerr.pl DO NOT EDIT
- * Copyright 2020-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2020-2024 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (paulidale) [inline providers/common/include/prov/proverr.h:3]: Drop this since there are no other changes to the file?

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
ping @openssl/committers

File: providers/common/include/prov/proverr.h (hunk 1 above)
Comment (slontis) [reply providers/common/include/prov/proverr.h:3]: Its a generated file..

**Comment by slontis** [comment]
Rebased

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**